tests/components/cover/test_command_line.py

```diff
@@ -44,41 +42,42 @@ def test_query_state_value(self):
 
     def test_state_value(self):
         """Test with state value."""
-        with tempfile.TemporaryDirectory() as tempdirname:
-            path = os.path.join(tempdirname, 'cover_status')
-            test_cover = {
-                'command_state': 'cat {}'.format(path),
-                'command_open': 'echo 1 > {}'.format(path),
-                'command_close': 'echo 1 > {}'.format(path),
-                'command_stop': 'echo 0 > {}'.format(path),
-                'value_template': '{{ value }}'
-            }
-            self.assertTrue(setup_component(self.hass, cover.DOMAIN, {
-                'cover': {
-                    'platform': 'command_line',
-                    'covers': {
-                        'test': test_cover
-                    }
+        test_cover = {
+            'command_state': 'cat cover_status',
+            'command_open': 'echo 1 > cover_status',
+            'command_close': 'echo 1 > cover_status',
+            'command_stop': 'echo 0 > cover_status',
+            'value_template': '{{ value }}'
+        }
+        self.assertTrue(setup_component(self.hass, cover.DOMAIN, {
+            'cover': {
+                'platform': 'command_line',
+                'covers': {
+                    'test': test_cover
                 }
-            }))
+            }
+        }))
 
-            state = self.hass.states.get('cover.test')
-            self.assertEqual('unknown', state.state)
+        with mock.patch('subprocess.call') as mock_call:
+            mock_call.return_value = True
 
             cover.open_cover(self.hass, 'cover.test')
-            self.hass.block_till_done()
-
-            state = self.hass.states.get('cover.test')
-            self.assertEqual('open', state.state)
-
             cover.close_cover(self.hass, 'cover.test')
-            self.hass.block_till_done()
-
-            state = self.hass.states.get('cover.test')
-            self.assertEqual('open', state.state)
-
             cover.stop_cover(self.hass, 'cover.test')
             self.hass.block_till_done()
 
-            state = self.hass.states.get('cover.test')
-            self.assertEqual('closed', state.state)
+            self.assertEqual(mock_call.call_count, 3)
+            self.assertEqual(
+                mock_call.call_args_list,
+                [
+                    mock.call(test_cover['command_open'], shell=True),
+                    mock.call(test_cover['command_close'], shell=True),
+                    mock.call(test_cover['command_stop'], shell=True)
+                ]
+            )
```

Comment:
Yeah, we should verify the states remain the same as they reflect if the call was succesful or not.